python/src/main/python/drivers/piglit-worker.py

```diff
@@ -248,34 +256,63 @@ def run_image_job(frag_file: str, json_file: str, status_file: str,
         graphicsfuzz_piglit_converter.main_helper(arglist)
     except Exception as ex:
-        runspv.log('Could not create shader_test from the given job.')
+        gfuzz_common.log('Could not create shader_test from the given job.')
         raise ex
-
-    status = STATUS_SUCCESS
-
-    shader_runner_cmd_list = shader_runner_cmd() + [shader_test_file, SHADER_RUNNER_ARG_AUTO,
-                                                    SHADER_RUNNER_ARG_UNIFORMS,
-                                                    SHADER_RUNNER_ARG_SUBTESTS]
+    shader_runner_cmd_list = shader_runner_cmd() + \
+        [shader_test_file, SHADER_RUNNER_ARG_AUTO,
+         SHADER_RUNNER_ARG_UNIFORMS, SHADER_RUNNER_ARG_SUBTESTS]
+    if use_catchsegv:
+        shader_runner_cmd_list.insert(0, catchsegv_cmd())
     if not skip_render:
         shader_runner_cmd_list.append(SHADER_RUNNER_ARG_PNG)
-    try:
-        runspv.subprocess_helper(shader_runner_cmd_list, timeout=TIMEOUT, verbose=True)
-    except subprocess.TimeoutExpired:
-        status = STATUS_TIMEOUT
-    except subprocess.CalledProcessError:
-        status = STATUS_CRASH
 
-    runspv.log('STATUS: ' + status)
+    gfuzz_common.remove(PNG_FILENAME)
+    gfuzz_common.remove(COMPARE_PNG_FILENAME)
+
+    status = \
+        gfuzz_common.run_catchsegv(shader_runner_cmd_list, timeout=TIMEOUT, verbose=True) \
+        if use_catchsegv else \
+        gfuzz_common.subprocess_helper(shader_runner_cmd_list, timeout=TIMEOUT, verbose=True)
 
     # Piglit throws the output PNG render into whatever the current working directory is
     # (and there's no way to specify a location to write to) - we need to move it to wherever our
     # output is.
 
-    if os.path.isfile(PNG_FILENAME):
-        shutil.move(PNG_FILENAME, png_file)
+    if not skip_render and status == STATUS_SUCCESS:
+        # An image was rendered, so we need to check for nondet. We do this by renaming the
+        # rendered image, rendering a second image, and using filecmp to compare the files.
+        if not os.path.isfile(PNG_FILENAME):
+            raise Exception("Shader runner successfully rendered, but no image was dumped?")
+        gfuzz_common.log('An image was rendered - rendering again to check for nondet.')
+        os.rename(PNG_FILENAME, COMPARE_PNG_FILENAME)
+        status = \
+            gfuzz_common.run_catchsegv(shader_runner_cmd_list, timeout=TIMEOUT, verbose=True) \
+            if use_catchsegv else \
+            gfuzz_common.subprocess_helper(shader_runner_cmd_list, timeout=TIMEOUT, verbose=True)
+        # Something is horribly wrong if shader crashes/timeouts are inconsistent per shader.
+        if not status == STATUS_SUCCESS:
+            raise Exception("Shader inconsistently fails - check your graphics drivers?")
+        if not os.path.isfile(PNG_FILENAME):
+            raise Exception("Shader runner successfully rendered, but no image was dumped?")
+        try:
```

Comment:
Please move the `try` up to the start of this block so that exception above will still cause the cleanup to be triggered.

Reply:
Done

Reply:
Done.